airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py

```diff
@@ -102,79 +128,303 @@ def get_updated_state(self, current_stream_state: MutableMapping[str, Any], late
+
+    @default_backoff_handler(max_tries=5, factor=5)
+    @user_defined_backoff_handler(max_tries=5)
+    def _send_request(self, request: requests.PreparedRequest, request_kwargs: Mapping[str, Any]) -> requests.Response:
+        response: requests.Response = self._session.send(request, **request_kwargs)
+        if self.should_retry(response):
+            custom_backoff_time = self.backoff_time(response)
+            raise UserDefinedBackoffException(backoff=custom_backoff_time, request=request, response=response)
+        else:
+            response.raise_for_status()
+
+        return response
+
     def _create_prepared_request(
-        self, path: str, headers: Mapping = None, params: Mapping = None, json: Any = None, data: Any = None
+        self, path: str, http_method: str = "GET", headers: Mapping = None, params: Mapping = None, json: Any = None, data: Any = None
     ) -> requests.PreparedRequest:
         """
-        Override to prepare request for AWS API.
-        AWS signature flow require prepared request to correctly generate `authorization` header.
-        Add `auth` arg to sign all the requests with AWS signature.
+        Override to make http_method configurable per method call
         """
-
-        return self._session.prepare_request(
-            requests.Request(method=self.http_method, url=self.url_base + path, headers=headers, params=params, auth=self._aws_signature)
+        args = {"method": http_method, "url": self.url_base + path, "headers": headers, "params": params}
+        if http_method.upper() in BODY_REQUEST_METHODS:
+            if json and data:
+                raise RequestBodyException(
+                    "At the same time only one of the 'request_body_data' and 'request_body_json' functions can return data"
+                )
+            elif json:
+                args["json"] = json
+            elif data:
+                args["data"] = data
+
+        return self._session.prepare_request(requests.Request(**args))
+
+    def _create_report(self) -> Mapping[str, Any]:
+        request_headers = self.request_headers()
+        replication_start_date = max(pendulum.parse(self._replication_start_date), pendulum.now("utc").subtract(days=90))
+        report_data = {
+            "reportType": self.name,
+            "marketplaceIds": self.marketplace_ids,
+            "createdSince": replication_start_date.strftime("%Y-%m-%dT%H:%M:%SZ"),
+        }
+        create_report_request = self._create_prepared_request(
+            http_method="POST",
+            path=f"{self.path_prefix}/reports",
+            headers=dict(request_headers, **self.authenticator.get_auth_header()),
+            data=json_lib.dumps(report_data),
         )
+        report_response = self._send_request(create_report_request, {})
 
-    def request_headers(self, *args, **kwargs) -> Mapping[str, Any]:
-        return {"content-type": "application/json"}
+        time.sleep(self.wait_seconds)
```

Comment:
why are we waiting a fixed amount instead of repeatedly checking if the report is done and if not, sleeping for some amount e.g: 

```
while not_done and total_time_waited <= MAX_TIMEOUT: 
    time.sleep(sleep_time)
```

if this is indeed the intended behavior could you add a comment explaining it?

Reply:
I thought it would be ok to use our backoff handler here. Added a feature you proposed